### DISCUSSION TITLE
Should crc cleaup remove everything except the config file?

### DISCUSSION THREAD

**Discussion by praveenkumar** [crc-org/crc#3208] [General]
Right now, `crc cleaup` do following.
- Try to remove all the modification which is part of `crc setup` (mostly related to network changes).
- Check the VM is exist or not and if there is then remove that using native hypervisor command.
- Remove `machines/crc` directory

What I think we should also add more action as part of cleanup, since this command is mostly used when user either had issue with setup/start or when try to upgrade and face issue.
- Remove cache directory.
     - Only drawback which I think is when user is having issue with same release and run cleanup then the bundle extraction will happen again (which is time consuming)
- Remove bin directory

- Keep the `crc.json` file which is the config file so user don't have to again fill same details using `crc config set` command.

**Comment by cfergeau** [comment]
That's a good question ;) However, using your rationale, I'd also remove the config file, because it can be a source of failed  `crc start`. I'd tend to keep the cache directory, unless we decide that `cleanup` should do `rm -rf ~/.crc`

**Comment by guillaumerose** [comment]
Proposition:
* Have a new command, `crc uninstall` which deletes *everything*
* Combine both `crc cleanup` and `crc setup` in a single command. 
 
If the users wants to remove crc, use `crc uninstall`. After that, he can trash `crc` binary.
If he wants to come back to clean state, use `crc setup` which run current cleanup then setup.

**Comment by cfergeau** [comment]
With current networking, this means undoing network configuration in `crc setup`, restarting networking, and then reconfiguring it (and restarting it again), this is a bit wasteful. But can be an interesting idea to revisit if we manage to remove the non-vsock code, and when we only ship installers on Windows/macOS (and no more standalone binaries).
